### PR TITLE
(console.log) Print className for an object if present

### PR DESCRIPTION
**Pull request by JibranKalia**
### What does this PR do?

This adds the functionality to print Object's class name if it exists matching Node's functionality. 

As well as adding adding tests for printing Maps, Sets and ArrayBuffers.

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [X] Code changes

### How did you verify your code works?

- [X] I or my editor ran Prettier on the changed files (or I ran `bun fmt`)
- [X] I checked the lifetime of memory allocated to verify it's (1) freed and (2) only freed when it should be
- [X] I or my editor ran `zig fmt` on the changed files
- [X] I included a test for the new code, or an existing test covers it

### CODE REVIEW

File: src/bun.js/bindings/exports.zig [hunk 1]
```diff
@@ -2862,6 +2862,12 @@ pub const ZigConsoleClient = struct {
                         .parent = value,
                     };
 
+                    var class_name = ZigString.Empty;
```
Comment (JibranKalia) [inline src/bun.js/bindings/exports.zig:2865]: Uses ZigString instead of bun.string because `getClassName` expects it.

**Comment by Jarred-Sumner** [comment]
Tests for `URLSearchParams` and `URL` seem to be failing

**Comment by JibranKalia** [comment]
> Tests for `URLSearchParams` and `URL` seem to be failing

Its hard for me to tell exactly what relevant tests failed in the Github actions UI. I ran `bun test url` locally and those tests passed. Do you have any tips?

**Comment by JibranKalia** [comment]
Just rebased on top of main. Hopefully, that's not an issue here. I usually do this at work but I am not sure if its frowned upon here.

**Comment by JibranKalia** [comment]
I think I will have to close this PR. I am not really able to figure out the issue. I got to here:

https://github.com/oven-sh/bun/blob/745b6b94ee56cad24d475799690cc9a89957d15b/src/bun.js/bindings/exports.zig#L1823-L1831

However, I can't understand why `value.getNameProperty` returns `URL` when passed the first child - "href". However, it doesn't do the that for an object instantiated from a Class.

**Comment by JibranKalia** [comment]
Closing this. Will try to figure out a different approach.

**Pull request closed without merging**